tensorboard/uploader/uploader.py

```diff
@@ -833,3 +836,35 @@ def _varint_cost(n):
         result += 1
         n >>= 7
     return result
+
+
+def _filter_graph_defs(events):
+    for e in events:
+        for v in e.summary.value:
+            if (
+                v.metadata.plugin_data.plugin_name
+                != graphs_metadata.PLUGIN_NAME
+            ):
+                continue
+            if v.tag == graphs_metadata.RUN_GRAPH_NAME:
+                data = v.tensor.string_val
+                data[:] = map(_filtered_graph_bytes, data)
+        yield e
+
+
+def _filtered_graph_bytes(graph_bytes):
+    try:
+        graph_def = graph_pb2.GraphDef().FromString(graph_bytes)
+    # The reason for the RuntimeWarning catch here is b/27494216, whereby
+    # some proto parsers incorrectly raise that instead of DecodeError
+    # on certain kinds of malformed input. Triggering this seems to require
+    # a combination of mysterious circumstances.
+    except (message.DecodeError, RuntimeWarning):
+        logger.warning(
+            "Could not parse GraphDef of size %d.", len(graph_bytes),
+        )
+        return graph_bytes
```

Comment:
Previously this effectively returned `None`, leading to the event being skipped entirely.  I think that's correct: the philosophy here is that we only bother uploading a graph that we will later be able to render.  Matching that will require adjusting the logic in `_filter_graph_defs` above, because uploading an empty blob is different from dropping the event.